packages/nodes-base/nodes/Zoho/GenericFunctions.ts

```diff
@@ -161,13 +167,18 @@ const omit = (propertyToOmit: string, { [propertyToOmit]: _, ...remainingObject
 /**
  * Place a product ID at a nested position in a product details field.
  */
-export const adjustProductDetails = (productDetails: ProductDetails) => {
+export const adjustProductDetails = (productDetails: ProductDetails, operation?: string) => {
 	return productDetails.map((p) => {
-		return {
-			...omit('product', p),
+		const adjustedProduct = {
 			product: { id: p.id },
 			quantity: p.quantity || 1,
 		};
+
+		if (operation === 'upsert') {
+			return adjustedProduct;
+		} else {
```

Comment:
For upsert operations this will only update product.id and quantity, but not fields like List Price etc.

Reply:
Back to the drawing board then 🤔